### ISSUE TITLE
DynamicRoute with getStaticPaths -> StaticRoute

### ISSUE THREAD

**Issue by sitch** [tatethurston/nextjs-routes#132]
For dynamic routes that export `getStaticPaths` it would be useful to have a config setting to enable automatic translation from `DynamicRoute` -> `StaticRoute`.

You definitely want this off by default in case it's an expensive operation

## Example

Give the page/pathname:
```ts
 "/blog/[slug]"
 ```
which exports `getStaticPaths`:
```ts 
const getStaticPaths = () => ({
  paths: ["1", "2", "3"]
})
```

Now, instead of just:

```ts
<DynamicRoute<"/blog/[slug]", { "slug": string }>
```

You get: 
```ts
<DynamicRoute<"/blog/[slug]", { "slug": string }>
<StaticRoute<"/blog/1">
<StaticRoute<"/blog/2">
<StaticRoute<"/blog/3">
```

### Note

One small detail is that if you have `fallback: false`

```ts 
const getStaticPaths = () => ({
  paths: ["1", "2", "3"],
  fallback: false
})
```
The `DynamicRoute` should be omitted:
```ts
<StaticRoute<"/blog/1">
<StaticRoute<"/blog/2">
<StaticRoute<"/blog/3">
```

**Comment by tatethurston** [comment]
Thanks @sitch. If there is sufficient interest in this I’ll look into implementation options. I’m curious to see what the next team’s solution for static sites will look like with the app directory and server components.

**Comment by sitch** [comment]
Sounds good -- I just saw this https://nextjs.org/blog/next-13-2#statically-typed-links

**Comment by tatethurston** [comment]
Yep, that only works for app directory (not pages) but if that works for your project LMK how it goes.
